Functions/Public/Get-GitHubGist.ps1

```diff
@@ -16,17 +16,34 @@ function Get-GitHubGist {
       
     .Link
     https://trevorsullivan.net
+    http://dotps1.github.io
     https://developer.github.com/v3/gists/
     #>
-    [CmdletBinding()]
-    param (
+    [CmdletBinding(DefaultParameterSetName = '__AllParameterSets')]
+    [OutputType([System.Object])]
+
+    Param (
+        [Parameter(ParameterSetName = 'Owner')]
+        [String]$Owner = (Get-GitHubAuthenticatedUser).login,
+        [Parameter(ParameterSetName = 'Id')]
+        [String]$Id,
+        [Parameter(ParameterSetName = 'Target')]
+        [ValidateSet('Public', 'Starred')]
+        [String]$Target
     )
     
-    $ApiCall = @{
-        Body = '';
-        RestMethod = '';
-        Method = '';
+    switch ($PSCmdlet.ParameterSetName) {
+        'Owner' { $restMethod = 'users/{0}/gists' -f $Owner; break; }
+        'Id' { $restMethod = 'gists/{0}' -f $Id; break; }
+        'Target' { if ($Target -eq 'Public') { $restMethod = 'gists/public'} else { $restMethod = 'gists/starred' }; break; }
+        default { $restMethod = 'gists'; break; }
+    }
+
+    $apiCall = @{
+        Body = ''
```

Comment:
If we don't have a body for the HTTP request, then we can just leave the parameter off the call to `Invoke-GitHubApi`.